e2e/test/scenarios/dashboard-filters/reproductions/20049-change-question-parameter-type-affects-dashboard-filter.cy.spec.js

```diff
@@ -0,0 +1,125 @@
+import { produce } from "immer";
+import { restore, visitDashboard, visitQuestion } from "e2e/support/helpers";
+import { SAMPLE_DATABASE } from "e2e/support/cypress_sample_database";
+import { SAMPLE_DB_ID } from "e2e/support/cypress_data";
+
+const { PRODUCTS } = SAMPLE_DATABASE;
+
+const filterDetails = {
+  name: "Category",
+  slug: "category",
+  id: "c32a49e1",
+  type: "string/contains",
+  default: ["Doohickey"],
+};
+
+const questionDetails = {
+  name: "Question",
+  native: {
+    query: "select * from products where {{category}} limit 10",
+    "template-tags": {
+      category: {
+        id: "6b8b10ef-0104-1047-1e5v-2492d5954555",
+        name: "category",
+        "display-name": "Category",
+        type: "dimension",
+        dimension: ["field", PRODUCTS.CATEGORY, null],
+        "widget-type": "string/contains",
+      },
+    },
+  },
+};
+
+const dashboardDetails = {
+  name: "Dashboard #20049",
+  parameters: [filterDetails],
+};
+const dashcardDetails = {
+  row: 0,
+  col: 0,
+  size_x: 16,
+  size_y: 8,
+};
+
+describe.skip("issue 20049", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+
+    cy.createNativeQuestionAndDashboard({
+      questionDetails,
+      dashboardDetails,
+    }).then(({ body: { id, card_id, dashboard_id } }) => {
+      cy.wrap(dashboard_id).as("dashboardId");
+      cy.wrap(card_id).as("cardId");
+
+      cy.request("PUT", `/api/dashboard/${dashboard_id}/cards`, {
+        cards: [
+          {
+            id,
+            card_id,
+            ...dashcardDetails,
+            parameter_mappings: [
+              {
+                card_id,
+                parameter_id: filterDetails.id,
+                target: ["dimension", ["template-tag", filterDetails.slug]],
+              },
+            ],
+          },
+        ],
+      });
+    });
+  });
+
+  it("Filter should stop applying if mapped question parameter is changed (metabase#20049)", () => {
+    cy.get("@cardId").then(cardId => {
+      updateQuestion(cardId);
+      visitQuestion(cardId);
+      verifyParameterType("String does not contain");
+    });
+
+    cy.get("@dashboardId").then(dashboardId => {
+      // visit dashboard again
+      visitDashboard(dashboardId);
+
+      // make sure parameter is not applied
+      cy.findAllByTestId("table-row").eq(0).should("not.contain", "Doohickey");
+    });
+  });
+});
+
+function updateQuestion(cardId) {
```

Comment:
I think this function is a bit confusing.
1. There are 2 `produce`s while we probably could use a single `produce` on the whole `questionDetails` itself to modify both `template-tags` and `parameters` at once.
2. The only part that's supposed to change on `parameters` is `type` not also `target` you won't need to set `target` if you have `parameters` in `questionDetails` in the first place ([example](https://github.com/metabase/metabase/pull/33632/files#diff-3dc802ccab554779b5b7944302d7e2f04c622b029fc3369168f81961fb168682R29-R37)). This way we really emphasize what needs to be changed

What do you think about making this change? I think it would simplify the code a lot more. It should look something like this.
```ts
  const newQuestionDetails = produce(questionDetails, draft => {
    draft.dataset_query["template-tags"][filterDetails.slug]["widget-type"] =
      "string/does-not-contain";

    draft.parameters[0].type = "string/does-not-contain";
  });
```

Reply:
good point, let me check. from the first sight I'd say we need two different objects, that's why I used `produce` twice

Reply:
@WiNloSt I tried this approach and it doesn't work the way I expect it to work. 
probably it's because of difference between questionDetails object and resulted question object (first has `native` key, second `dataset_query`). Also I put those updated objects into different places, so I'd have them separately for simplicity

Reply:
@uladzimirdev Let me try something out. Probably after our next meeting. I think I know how to make it work.

Reply:
I had very similar code in my local version, thanks for finishing it!